src/gui/processing/models/qgsmodelcomponentgraphicitem.cpp

```diff
@@ -890,6 +890,44 @@ void QgsModelChildAlgorithmGraphicItem::contextMenuEvent( QGraphicsSceneContextM
       QAction *deactivateAction = popupmenu->addAction( QObject::tr( "Deactivate" ) );
       connect( deactivateAction, &QAction::triggered, this, &QgsModelChildAlgorithmGraphicItem::deactivateAlgorithm );
     }
+
+    // only show the "View Output Layers" action for algorithms which create layers
+    if ( const QgsProcessingAlgorithm *algorithm = child->algorithm() )
+    {
+      const QList< const QgsProcessingParameterDefinition * > outputParams = algorithm->destinationParameterDefinitions();
+      if ( !outputParams.isEmpty() )
+      {
+        popupmenu->addSeparator();
+        QAction *viewOutputLayersAction = popupmenu->addAction( QObject::tr( "View Output Layers" ) );
+        viewOutputLayersAction->setIcon( QgsApplication::getThemeIcon( QStringLiteral( "mActionShowSelectedLayers.svg" ) ) );
```

Comment:
Not sure if the icon is the best option here, but I've got no better suggestion for now.